src/org/anjocaido/groupmanager/data/DataUnit.java

```diff
@@ -43,7 +44,7 @@
 	private boolean changed;
 	private long timeStamp = 0;
 
-	private Map<String, Long> permissions = Collections.synchronizedSortedMap(new TreeMap<>(new StringPermissionComparator()));
+	private final Map<String, Long> permissions = Collections.synchronizedSortedMap(new TreeMap<>(new StringPermissionComparator()));

```

Comment:
I don't have an environment where I can do prolonged and extensive testing, so either this would need to be reverted or it would be published as a potentially highly unstable build